### PR TITLE
Implement alternative duplicate markers

### PR DESCRIPTION
**Pull request by pontushojer**
Here is the implementation of sambamba and samblaster as duplicate markers. Picard MarkDuplicates has been removed as it was proven to be quite resource intensive and slow (see #113).

Note that samblaster is integrated as pipe in the mapping step as this is the intended benefit of the tool. Thus it will skip the `mark_duplicates` step and instead go from `tagbam` directly to `clusterrmdup`. The naming is a bit off as the file `mapped.sorted.bam` now has duplicates marked. I think this is fine for now and we can change it whenever we select a specific tool for marking duplicates. But comment if you have any objections or suggestions!

### CODE REVIEW

File: src/blr/Snakefile [hunk 1]
```diff
@@ -96,18 +95,27 @@ rule map_reads:
...
         }
         command = commands[config["read_mapper"]].format(**locals(), **globals())
-        shell(
-            "{command} 2> >(tee {log} >&2) |"
-            "samtools sort -"
-            " -@ {threads}"
-            " -O BAM > {output.bam}"
-        )
+        if config["duplicate_marker"] == "samblaster":
+            shell(
```
Comment (marcelm) [inline src/blr/Snakefile]: You can avoid the duplication of the entire shell command by setting a variable to `"samblaster |"` if the configured duplicate marker is samblaster and to `""` otherwise. Then just use that variable in the string that is passed to the shell command.

Comment (pontushojer) [reply src/blr/Snakefile]: That is true 👍

File: src/blr/blr.yaml [hunk 2]
```diff
@@ -4,7 +4,8 @@ molecule_tag: MI  # Used to store molecule ID, same as 10x default.
 num_mol_tag: MN # Used to store number of molecules per barcode
 sequence_tag: RX    # Used to store original barcode sequence in bam file. 'RX' is 10x genomic default
 genome_reference:  # Path to indexed reference
-read_mapper: bowtie2  # Choose bwa, bowtie2 or minimap2
+read_mapper: bowtie2  # Choose bwa or bowtie2 or minimap2
```
Comment (marcelm) [inline src/blr/blr.yaml]: Hm, why did you change the grammar here? This was quite standard English before.

Comment (pontushojer) [reply src/blr/blr.yaml]: I think I made a mistake while rebasing, will change.

File: src/blr/Snakefile [hunk 3]
```diff
@@ -123,15 +131,13 @@ rule mark_duplicates:
...
-        stderr = "picard_mkdup.log"
+    log: "mark_duplicates.log"
+    threads: 20
     shell:
-        "picard -Xms{config[heap_space]}g MarkDuplicates"
-        " I={input.bam}"
-        " O={output.bam}"
-        " M={log.metrics}"
-        " ASSUME_SORT_ORDER=coordinate 2> {log.stderr}"
+        "sambamba markdup "
```
Comment (marcelm) [inline src/blr/Snakefile]: Please remove the space before the last `"` in this and the next line.

Comment (pontushojer) [reply src/blr/Snakefile]: Fixed!

**Comment by FrickTobias** [comment]
Loos great!
